mobile/mysterium/openvpn_connection_setup.go

```diff
@@ -29,22 +29,40 @@ import (
 	"github.com/mysteriumnetwork/node/services/openvpn/session"
 )
 
+type openvpn3SessionFactory func(connection.ConnectOptions) (*openvpn3.Session, error)
+
+var errSessionWrapperNotStarted = errors.New("session wrapper not started")
+
 type sessionWrapper struct {
-	session *openvpn3.Session
+	session        *openvpn3.Session
+	sessionFactory openvpn3SessionFactory
```

Comment:
Type name is okeysh, but sessionFactory -> createSession. Later in code session, err := wrapper.createSession(options) which is more readable